craft_parts/packages/base.py

```diff
@@ -64,7 +64,9 @@ def refresh_build_packages_list(cls) -> None:
 
     @classmethod
     @abc.abstractmethod
-    def install_build_packages(cls, package_names: List[str]) -> List[str]:
+    def install_build_packages(
+        cls, package_names: List[str], list_only: bool = False
```

Comment:
The dry-run / list-only functionality should be its own method which install_build_packages() uses.

Reply:
I like the idea, I'll add a note for implementation in the big repo refactoring.